modules/doc/content/newsletter/2021_06.md

```diff
@@ -2,3 +2,5 @@
 
 
```

Comment:
## Multiple input file support

Users can now supply multiple input file names after the `-i` option. The supplied inputs are successively merged into one simulation specification, with later files adding and overwriting parameters. This allows users to factor out common parameters (e.g. solver options, postprocessors, outputs) from a set of similar input files

 ```
 ./mooseapp-opt -i common_parameters.i simulation_1.i 
```

The output file names will be based off the _last_ input file name specified (unless the `Outputs/file_base` parameter is specified). When multiple inputs are specified, MOOSE will inform the user about parameter overrides at the top of the simulation output. 

To assist users with refactoring input files several new functions we added to the `hit` commandline tool.

- `hit merge -output outfile.i infile1.i infile2.i ...` will produce a single file `outfile.i` that is a merge of the `infile*.i` inputs (which would result in the same MOOSE simulation as the multiple input files).
- `hit diff left.i right.i` or `hit diff -left left1.i left2.i ... -right right1.i right2.i ...` performs a diff on the left and right files (in the second example merging all left and right files respectively first). This diff is not sensitive to order, formatting, or comments in the input files.
- `hit common file1.i file2.i ... > common.i` will extract all parameters that are common to the specified input files (and have the same values). This can be used as the first step in factoring out common settings into a single input file.
- `hit subtract common.i simulation_1_full.i > simulation_1.i` removes the parameters in `common.i` from `simulation_1_full.i`, creating `simulation_1.i`. This is the second step in factoring out common parameters from a set of input files. The resulting file can be run as `./mooseapp-opt -i common.i simulation_1.i` and will result in teh same simulation as `./mooseapp-opt -i simulation_1_full.i`.

Reply:
am I supposed to make a PR to your repo @aeslaughter ?

Reply:
members of the CCB can just push directly to my branch, non members need to make a PR to my repo

Reply:
Well, I'm a CCB member, but I get

```
schwd$ git push --set-upstream aeslaughter  mooseletter-11447
ERROR: Authentication error: Authentication required: You must have push access to verify locks
error: failed to push some refs to 'github.com:aeslaughter/moose'
```

Reply:
You can cherry-pick c76c01d06f8d8c9296c139bbdc5e6a35e5921a40

Reply:
You are doing something wrong. I usually do something like `git push aeslaughter HEAD:mooseletter-11447` 

![Screen Shot 2021-07-12 at 8 48 03 AM](https://user-images.githubusercontent.com/341333/125308394-1b1e5a80-e2ee-11eb-988d-7cb779fb4fdf.png)

Reply:
```
(moose) fn606041:moose schwd$ git push aeslaughter HEAD:mooseletter-11447
ERROR: Authentication error: Authentication required: You must have push access to verify locks
error: failed to push some refs to 'github.com:aeslaughter/moose'
```

in my `.git/config` I have the proper ssh git url:
```
(moose) fn606041:moose schwd$ git push aeslaughter HEAD:mooseletter-11447
ERROR: Authentication error: Authentication required: You must have push access to verify locks
error: failed to push some refs to 'github.com:aeslaughter/moose'
```

Reply:
OK, I will grab your commit.

Reply:
@dschwen I was able to push to @aeslaughter 's branch this morning